### PR TITLE
streamingccl: add a builtin function that retrives stream ingestion statitics

### LINKED ISSUE #57430: streaming: make cluster-to-cluster streaming observable
This is a parent issue that tracks discussion and progress on implementing metrics and observability tooling around cluster streaming.

- [x] [@pbardea] Update job progress payload to contain a map of per-progress partition information (resolved timestamp, most recent error, etc) (https://github.com/cockroachdb/cockroach/pull/68338)
- [ ] Expose builtin that exposes per-partition information via SQL
- [x] Add Prometheus metrics (https://github.com/cockroachdb/cockroach/pull/68982)

Jira issue: CRDB-3538

Epic CRDB-6403

### PR DESCRIPTION
**Pull request by gh-casper**
Support crdb_internal.stream_ingestion_stats(ingestion_job_id) to
running progress for a stream ingestion job.

Release note (sql change): Support crdb_internal.stream_ingestion_stats
(ingestion_job_id) to running progress for a stream ingestion job.

Release Justification: Cat 4.

Closes: https://github.com/cockroachdb/cockroach/issues/57430

### CODE REVIEW

File: pkg/ccl/streamingccl/streamproducer/stream_lifetime.go [hunk 1]
```diff
@@ -289,3 +290,30 @@ func completeReplicationStream(
 			return nil
 		})
 }
+
+func getStreamIngestionStats(
+	evalCtx *tree.EvalContext, txn *kv.Txn, ingestionJobID jobspb.JobID,
+) (*streampb.StreamIngestionStats, error) {
+	registry := evalCtx.Planner.ExecutorConfig().(*sql.ExecutorConfig).JobRegistry
+	j, err := registry.LoadJobWithTxn(evalCtx.Ctx(), ingestionJobID, txn)
```
Comment (miretskiy) [inline pkg/ccl/streamingccl/streamproducer/stream_lifetime.go:298]: Do we need to use withTxn variety?

**Review by miretskiy — APPROVED** [review]

File: pkg/sql/sem/builtins/replication_builtins.go [hunk 2]
```diff
@@ -70,6 +73,45 @@ var replicationBuiltins = map[string]builtinDefinition{
...
+				}
+				jsonDatum, err := tree.ParseDJSON(string(jsonStats))
+				if err != nil {
+					return nil, err
+				}
+				return jsonDatum, nil
+			},
+
+			Info: "This function can be used on the ingestion side to get a statistics summary in json format " +
+				"of a stream ingestion job",
```
Comment (otan) [inline pkg/sql/sem/builtins/replication_builtins.go:109]: nit: full stop at the end here

**Review by otan — APPROVED** [review]
nit: commit title / pr title misspelling of statistics

**Pull request merged**